### PR TITLE
Update install instructions in 2.x docs

### PR DESCRIPTION
**Pull request by discdiver**
### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [ ] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,12 +33,14 @@ Every activity is tracked and can be monitored with a self-hosted [Prefect serve
 
 ## Getting started
 
-Prefect requires Python 3.8 or later. To [install Prefect](https://docs.prefect.io/getting-started/installation/), run the following command:
+Prefect requires Python 3.8 or later. To [install Prefect 2](https://docs-2.prefect.io/getting-started/installation/), run the following command:
 
 ```bash
-pip install prefect
+pip install prefect<3
```
Comment (zzstoatzz) [inline README.md:39]:
```suggestion
pip install 'prefect<3'
```

File: docs/contributing/overview.md [hunk 2]
```diff
@@ -12,16 +12,18 @@ search:
...
 ## Setting up a development environment
 
 First, download the source code and install an editable version of the Python package:
 
 <div class="terminal">
 ```bash
-# Clone the repository
-git clone https://github.com/PrefectHQ/prefect.git
+# Clone the repository's 2.x branch
+git clone --branch 2.x https://github.com/PrefectHQ/prefect.git
```
Comment (zzstoatzz) [inline docs/contributing/overview.md:26]: they wouldn't be committing to 2.x directly, so i think this is unnecessary, we should make a more normal recommendation
```suggestion
git clone https://github.com/PrefectHQ/prefect.git
```

File: docs/contributing/overview.md [hunk 3]
```diff
@@ -12,16 +12,18 @@ search:
...
 cd prefect
```
Comment (zzstoatzz) [inline docs/contributing/overview.md:27]:
```suggestion
cd prefect
git checkout -b new-branch-2.x 2.x
```

File: docs/getting-started/installation.md [hunk 4]
```diff
@@ -39,22 +39,24 @@ Learn more about Prefect Cloud solutions for enterprises [here](https://www.pref
...
+### Installing the Prefect 2 latest version
 
 Prefect is published as a Python package.
-To install the latest release or upgrade an existing Prefect install, and upgrade existing Python dependencies, run the following command in your terminal:
+To install the latest Prefect 2 release run the following command in your terminal:
 
 <div class="terminal">
 ```bash
-pip install -U prefect
+pip install prefect<3
```
Comment (zzstoatzz) [inline docs/getting-started/installation.md:49]:
```suggestion
pip install 'prefect<3'
```

File: docs/getting-started/installation.md [hunk 5]
```diff
@@ -139,11 +141,11 @@ The `prefect-client` library is a minimal installation of Prefect designed for i
 `prefect-client` enables a subset of Prefect's functionality with a smaller installation size, making it ideal for use in lightweight, resource-constrained, or ephemeral environments.
 It omits all CLI and server components found in the `prefect` library.
 
-Install the latest version with:
+Install with:
 
 <div class="terminal">
 ```bash
-pip install -U prefect-client
+pip install prefect-client<3
```
Comment (zzstoatzz) [inline docs/getting-started/installation.md:148]:
```suggestion
pip install 'prefect-client<3'
```

File: docs/getting-started/installation.md [hunk 6]
```diff
@@ -216,15 +218,15 @@ pip3 install prefect
 
 ## Installing unreleased code
 
-To use the most up-to-date, unreleased Prefect code, you can install directly off the `main` GitHub branch:
+To use the most up-to-date, unreleased Prefect 2 code, you can install directly off the `2.x` GitHub branch:
 
 <div class="terminal">
 ```bash
-pip install -U git+https://github.com/PrefectHQ/prefect
+pip install --branch 2.x -U git+https://github.com/PrefectHQ/prefect
```
Comment (zzstoatzz) [inline docs/getting-started/installation.md:225]: same as above

File: docs/tutorial/index.md [hunk 7]
```diff
@@ -48,7 +48,7 @@ Then install Prefect:
 
 <div class="terminal">
 ```bash
-pip install -U prefect
+pip install prefect<3
```
Comment (zzstoatzz) [inline docs/tutorial/index.md:51]:
```suggestion
pip install 'prefect<3'
```

**Review by zzstoatzz — CHANGES_REQUESTED** [review]
some syntax corrections and a comment on the `git` commands

File: docs/getting-started/installation.md (hunk 6 above)
Comment (discdiver) [reply docs/getting-started/installation.md]: Removing this section as users can find in the contributing section.

**Review by zzstoatzz — APPROVED** [review]

**Pull request merged**